datalad/distribution/update.py

```diff
@@ -97,41 +108,18 @@ def __call__(
         if fetch_all is not None:
             lgr.warning('update(fetch_all=...) called. Option has no effect, and will be removed')
 
-        if not dataset and not path:
-            # try to find a dataset in PWD
-            dataset = require_dataset(
-                None, check_installed=True, purpose='updating')
+        refds = require_dataset(dataset, check_installed=True, purpose='updating')
         refds_path = Interface.get_refds_path(dataset)
```

Comment:
`refds_path` can now be None, which would lead to None being used for `refds` in the result dictionary and lead to the refds path not being filtered from `save_paths` before passing it to `refs.save` (though `.save` handles it OK).  Perhaps this should just be `refds_path = refds.path`?

Reply:
Done, thx.